lightning/src/ln/channel.rs

```diff
@@ -843,16 +846,24 @@ impl<Signer: Sign> Channel<Signer> {
 	}
 
 	/// Returns a minimum channel reserve value the remote needs to maintain,
-	/// required by us.
+	/// required by us according to the configured or default
+	/// [`ChannelHandshakeConfig::their_channel_reserve_proportion_millionth`]
 	///
 	/// Guaranteed to return a value no larger than channel_value_satoshis
 	///
-	/// This is used both for new channels and to figure out what reserve value we sent to peers
-	/// for channels serialized before we included our selected reserve value in the serialized
-	/// data explicitly.
-	pub(crate) fn get_holder_selected_channel_reserve_satoshis(channel_value_satoshis: u64) -> u64 {
+	/// This is used both for outbound and inbound channels and has lower bound of `MIN_THEIR_CHAN_RESERVE_SATOSHIS`
+	pub(crate) fn get_holder_selected_channel_reserve_satoshis(channel_value_satoshis: u64, config: &UserConfig) -> u64 {
+		let calculated_reserve = channel_value_satoshis.saturating_mul(config.channel_handshake_config.their_channel_reserve_proportion_millionth as u64) / 1_000_000;
+		cmp::min(channel_value_satoshis, cmp::max(calculated_reserve, MIN_THEIR_CHAN_RESERVE_SATOSHIS))
+	}
+
+	/// This is for legacy reasons, present for forward-compatibility. LDK versions older than " < 0.0.104"
+	/// don't know how read/handle values other than default from storage. Hence we use this function
```

Comment:
Yeah fair enough :)! I agree on that my suggestion isn't correct either.

What I specifically find confusing is the sentence:
"LDK versions older than " < 0.0.104" don't know how read/handle values other than default from storage.",
as those versions under no circumstances read the value from the storage at all, not even the default value. But, given the discussion about being too nit picky above, maybe that applies to me here as well 😁.

If @G8XSU thinks changing the wording here would clarify the docs, feel free to change this. Either way, I'm ok with the current wording here as well :)!

Reply:
Main purpose: Its present for forward-compatibility, so that we don't persist default values for channels created by older versions. This is for the case when we have to roll-back to previous version, we have at-least those old channels operational.

If in future, we decide to remove forward compat with <104, we can get rid of this function/logic. 
I think this information is conveyed.
Didn't want to make it anymore complex than that.